Simplification of scaffold structure and removing docker assets

grieve opened the pull request:
Restructure and simplification of a lot of components. All docker assets are removed in favour of automatically built containers from docker hub.

cyberphobia committed:
Added test to check inheritance of handlers.

pames committed:
Merge pull request #2 from insgadget/master

Added test to check inheritance of handlers.

Commit:
Remove duplicated command @ Dependency Setup

pames committed:
Merge pull request #3 from tavancini/typo_fix

Remove duplicated command @ Dependency Setup

JustinAtPsycle committed:
Refactor Jinja2 implementation in BaseHandler

Leverage webapp2_extras.jinja2, caching the Jinja2 instance in app registry. Creating
an instance on each request can lead to to increased latency under load. Keep config of Jinja2
instance in BaseHandler rather than in webapp2 config (main.py)

pames committed:
Merge pull request #4 from JustinAtPsycle/jinja2-refactor

Jinja2 performance improvement refactor

pames committed:
Add test to force routes that use handler_method to explicitly declare which verbs they service.

pames committed:
Merge pull request #5 from pames/handler_method

Add test to force routes that use handler_method to explicitly declare which verbs they service

insgadget committed:
Add MultiRoute support to main_test.py

pames committed:
Merge pull request #7 from insgadget/master

Add MultiRoute support to main_test.py

Commit:
Fixing autoescape bug and moving sample handlers to separate directory

pames committed:
Merge pull request #11 from carstein/master

Fixing autoescape bug and moving sample handlers to separate directory

pames committed:
Shallow copy CSP Policy dictionary prior to modification when rendering headers.

lweichselbaum committed:
Harden default CSP

pames committed:
Merge pull request #16 from lweichselbaum/harden-csp

Harden default CSP

pames committed:
Merge pull request #15 from pames/i14

Copy CSP Policy dict prior to modification when rendering

Commit:
removing js third party from server src

Commit:
moving client and server folders to nest under src

Commit:
clean up and simplification

Commit:
fixing docker compose service for client

Commit:
switching default FE code to use node container

grieve commented:
@pemcconnell @rickydunlop @mjmcconnell 

Just tagging to cast your eyes over this. It's a big push but it's mostly just docker changes and a folder restructure. Key changes are:

- `src-client` and `src-server` are now `src/client` and `src/server` as it's cleaner and makes FE gulp work nice and consistent.
- **all** docker building has been removed, we use the automated build of our appengine-dev container from docker hub and just use a plain node container for the FE build.
- makefile has been greatly simplified since no docker building is required
- lots of ops files pertaining to docker have been removed
- docker-compose now orchestrates both containers

rickydunlop commented on ops/docker-compose.yml:
```diff
@@ -0,0 +1,21 @@
+    - 8000:8000
+
+gaescaffoldclient:
+  image: node:5.0.0
```
5.4.1 is the latest stable release, what's the thoughts on using just node:5 or node:5.4 instead of 5.0.0?
See https://hub.docker.com/_/node/ for reference

grieve replied:
Sure - I'd go with 5.4 given our latest experiences with minor bumps :smirk:

rickydunlop commented on src/client/package.json:
```diff
@@ -6,5 +6,9 @@
     "gulp-watch": "^3.0.0"
   },
+  "scripts": {
+    "build": "gulp",
```
Maybe rename this to 
`"gulp": "gulp",`
as this allows us to use gulp for other gulp tasks too.
`npm gulp <task>`

grieve replied:
It's demonstrative only - the fe skeleton always replaces this code, or another stack - this is just an opinionated demo of the structure.

grieve replied:
(And the FE skeleton has now changed to `npm run gulp` anyway)

Commit:
bumping node image to 5.4

rickydunlop commented:
I'm happy for this to be merged if everyone else is ok with it

mjmcconnell commented:
Yea, looks good, though currently has conflicts. Guessing my test_wsgi fix broke it

miuraken committed:
make --dev option valid

pames committed:
Merge pull request #19 from miuraken/grunt-fix

make --dev option valid in Gruntfile.js

lweichselbaum committed:
Add strict-csp support

Commit:
Merge branch 'master' into develop

pames committed:
Merge branch 'master' of https://github.com/lweichselbaum/gae-secure-scaffold-python into lweichselbaum-master

pames committed:
Merge branch 'lweichselbaum-master' / PR24

rickydunlop committed:
Use latest node version

Commit:
Fixing an issue where '-dev' was always being added to versionString when
deploying to AppEngine on a machine using Git v2.9.1 or higher.

Details:
From Git v2.9.1, the messaging when there are no uncommitted changes has changed.
See: https://github.com/git/git/blob/e0c1ceafc5bece92d35773a75fff59497e1d9bd5/Documentation/RelNotes/2.9.1.txt#L53

The PR includes a fix so that both the old and new message will be tested
against, and the -dev will be added as intended.

n1ruragu committed:
Soy Py Support

Update Gruntfile.js

Update Gruntfile.js

Update Gruntfile.js

pames committed:
Merge pull request #27 from n1ruragu/soypy-fixes

Soy Python Support

pames committed:
Add/update closure-{compiler,library,templates}.

pames committed:
Update npm dependencies.

pames committed:
Improved support for Python Closure Templates.

pames committed:
Merge pull request #25 from joewanko/to-master/10-19-version-fix

Update git pattern matching for unmodified repositories.

pames committed:
Merge pull request #29 from pames/soy

First class support for Python Closure Templates

mikispag committed:
Add base-uri 'self' to CSP.

pames committed:
Merge pull request #31 from mikispag/master

Add base-uri 'self' to CSP.

Commit:
removing circle

Commit:
Merge branch 'master' of github.com:google/gae-secure-scaffold-python into release/2017-ff-scaffold

Commit:
updating to meet tests

Commit:
ensuring all templates and examples work as expected and better jinja2 config

grieve committed:
Merge pull request #12 from rehabstudio/release/2017-ff-scaffold

Release/2017 ff scaffold

grieve commented:
Closing due to rigor mortis.
